programs/uxd/src/instructions/lsd/initialize_lsd_depository.rs

```diff
@@ -0,0 +1,153 @@
+use anchor_lang::prelude::*;
+use anchor_spl::token::Mint;
+use anchor_spl::token::Token;
+use anchor_spl::token::TokenAccount;
+
+use crate::LSD_PROFITS_TOKEN_ACCOUNT_NAMESPACE;
+use crate::error::UxdError;
+use crate::events::InitializeLsdDepositoryEvent;
+use crate::state::LSD_DEPOSITORY_SPACE;
+use crate::state::LsdDepository;
+use crate::validate_is_program_frozen;
+use crate::Controller;
+use crate::CONTROLLER_NAMESPACE;
+use crate::DEFAULT_REDEEMABLE_UNDER_MANAGEMENT_CAP;
+use crate::LSD_DEPOSITORY_ACCOUNT_VERSION;
+use crate::LSD_DEPOSITORY_NAMESPACE;
+
+#[derive(Accounts)]
+pub struct InitializeLsdDepository<'info> {
+    /// #1 Authored call accessible only to the signer matching Controller.authority
+    pub authority: Signer<'info>,
+
+    /// #2
+    #[account(mut)]
+    pub payer: Signer<'info>,
+
+    /// #3
+    #[account(
+        mut,
+        seeds = [CONTROLLER_NAMESPACE],
+        bump = controller.load()?.bump,
+        has_one = authority @UxdError::InvalidAuthority,
+    )]
+    pub controller: AccountLoader<'info, Controller>,
+
+    /// #4 
+    #[account(
+        init,
+        seeds = [LSD_DEPOSITORY_NAMESPACE, collateral_mint.key().as_ref()], 
+        bump,
+        payer = payer,
+        space = LSD_DEPOSITORY_SPACE,
+    )]
+    pub depository: AccountLoader<'info, LsdDepository>,
+
+    /// #5
+    pub collateral_mint: Account<'info, Mint>,
+
+    /// #6
+    pub liquidation_mint: Account<'info, Mint>,
+
+    /// #7
+    #[account(
+        init,
+        seeds = [LSD_PROFITS_TOKEN_ACCOUNT_NAMESPACE, depository.key().as_ref()],
+        token::authority = depository,
+        token::mint = liquidation_mint,
+        bump,
+        payer = payer,
+    )]
+    pub profits_token_account: Account<'info, TokenAccount>,
```

Comment:
if we design to have a borrow fee settings here, 
then it also require a token account storing the redeemable that the user pay upon borrowing?
And same with repay fee.
So in total there would be `redeemable_profits_token_account`, `collateral_profits_token_account`, `liquidation_profits_token_account`

Reply:
Just thinking based on my reading of the rest of the PR:
- this ATA is probably going to be in "liquidation_mint" amount so more like "profits_liquidation"
- the user most likely will pay fees using collateral rather than redeemable right ? or we just mint "less" redeemable ? I dont know if we need to store any redeemable inside of the depository

Reply:
Depends on how we wanna store fee as, but like for solend if there are a borrow fee, then its collected on the borrowing mint which is redeemable on our case, like if u are borrowing 10k UXD and borrowing fee on 0.05% then 10k  is minted but u only take home 9995, and we take the 5 remaining. To repay the whole position u have to mint/swap elsewhere 5 more. 
But agree "mint less" is a easier way.

Reply:
Yhea, for now I factored it all into a profits token account and I left a comment somewhere "swap to profit mint" on collection